json_serialization/reader.nim

```diff
@@ -97,7 +97,7 @@ proc assignLineNumber*(ex: ref JsonReaderError, r: JsonReader) =
   ex.line = r.lexer.line
   ex.col = r.lexer.tokenStartCol
 
-proc raiseUnexpectedToken*(r: JsonReader, expected: ExpectedTokenCategory)
+proc raiseUnexpectedToken*(r: var JsonReader, expected: ExpectedTokenCategory)
```

Comment:
How is the reader modified here?

Reply:
`r.lexer.tok` needs a call by ref entry as it can resolve a lazy token -- well it make sense to use a `lazyTok` here